### ISSUE TITLE
Localise the 'Welcome' article and other default content (such as category texts)

### ISSUE THREAD

**Issue by philwareham** [textpattern/textpattern#1179]
I'm pretty sure we can leverage the recent installation changes and make the various default content installed at setup more localised. I can put the content onto Crowdin for translation to expedite this.

This is especially true of things like RTL languages and other non-Latin alphabets where we have an English article amongst the already localised content. Not a great advert for language support!

**Comment by philwareham** [comment]
Article, category, links and section translation XML files are now on Crowdin for people to translate. We can sort out how it integrates into the setup process later.

**Comment by philwareham** [comment]
OK, I've created [a branch](https://github.com/textpattern/textpattern/tree/localised-setup-content) for this work to reside in, and put some translated content in there as a starting point. The setup process will be broken in this branch for now.

Things to consider:

1. The structure of how we house the localised content. For now I've done it like this:

```
setup
-- initial-content
---- en
------ articles
-------- articles.welcome.xml
------ data
-------- core.prefs
-------- txp_category.xml
-------- txp_link.xml
-------- txp_section.xml
---- fi
------ articles
-------- articles.welcome.xml
------ data
-------- core.prefs
-------- txp_category.xml
-------- txp_link.xml
-------- txp_section.xml
...
```

Better suggestions welcome. e.g. I don't know if we *need* the `articles` and `data` subfolders or not really.

2. Whether we allow the section name (not title) to be localised too, or if this causes problems. It would provide localised URL paths which is obvious better, but is this a step too far?

Obviously I can't do the PHP coding behind this myself :) but I hope it'd be relatively simple for a willing helper.

**Comment by petecooper** [comment]
@philwareham - like this a lot.

1. This looks great. I would perhaps err towards `default-content` rather than `initial-content` but I'm verging on nitpicking.

Perhaps some sort of JSON package identifier in the root for a bit of future-proofing?:

```
setup
-- initial-content
---- en
------ articles
-------- articles.welcome.xml
------ data
-------- core.prefs
-------- txp_category.xml
-------- txp_link.xml
-------- txp_section.xml
------ package.json
---- fi
------ articles
-------- articles.welcome.xml
------ data
-------- core.prefs
-------- txp_category.xml
-------- txp_link.xml
-------- txp_section.xml
------ package.json
```

2. I vote +1 for translating the section name _and_ title. It seems counterintuitive to me to go this far and not top-and-tail it. If we can avoid URLs like `https://пример.com/articles/надежды-на-будущее` then that'd be _super_.

3. All of the above makes more translation work, but broadens the audience with native language support from the outset.

**Comment by phiw13** [comment]
Question: how do you plan to handle *inline textile* (links, `strong`, `em`…) in CJK languages (and perhaps other languages) where Textile fails ?

**Comment by petecooper** [comment]
>Question: how do you plan to handle inline textile

Presumably standard HTML would be a fallback if Textile can't be cajoled into accepting multibyte or non-Latin character sets fully.

**Comment by phiw13** [comment]
> Presumably standard HTML would be a fallback

Yes, that is what we use for Japanese. (MarkDown works fine, FWIW)

**Comment by philwareham** [comment]
If Textile can't be fixed to allow this, then maybe we should be careful to make sure the welcome article overall doesn't contain and breaking Textile syntax, or tailor each non-latin language article accordingly. Open to suggestions here.

**Comment by philwareham** [comment]
@petecooper can you give an example of what the JSON file would contain?

**Comment by petecooper** [comment]
@philwareham - ToC for the language, whether it's formatted in Textile, Markdown, HTML or other, whether the article is translated or not, generated checksum for the stuff that's going to be imported (perhaps to mitigate against https://github.com/textpattern/textpattern/issues/1141 or similar), that sort of thing. It's feasible that all this can be included in the files rather than an associate description, of course - it was more of a future-proofing suggestion than anything else.

**Comment by petecooper** [comment]
>If Textile can't be fixed to allow this, then maybe we should be careful to make sure the welcome article overall doesn't contain and breaking Textile syntax, or tailor each non-latin language article accordingly?

Perhaps a wild thought…

Working on the premise that a) many or most people will simply delete the initial article after its useful life (i.e. "hey, here are some links"), and won't master Textile based on one article with basic markup, does that default article _need_ to be included as Textile?

It could be construed as perhaps being a missing vote of confidence if the default stuff is in HTML, but from a practicality standpoint, more translators will understand basic HTML than basic Textile.

**Comment by petecooper** [comment]
>can you give an example of what the JSON file would contain?

Actually, there's nothing I can think of that has to live outside the various XML files, so bin the idea of a JSON file for future proofing. Can be easily added in the, erm, future.

**Comment by petecooper** [comment]
Since we're making this article and the default fixtures and fittings more global-ish, I am trying to think objectively and fairly about _all_ languages, and lower/remove any hurdles that translators might encounter.

"My pithy slogan" has sometimes struck me as a bit clever, or smug. It might also be mistaken for the white stuff inside oranges and lemons. So, yeah.

How about something more in-tune with the tone of the welcome article:

* "My site slogan"
* "My website slogan"

…or something else entirely.

**Comment by philwareham** [comment]
Agreed, I've always found that line a bit pompous too. 'My site slogan" works for me (as we already say 'My site' for default website name).

**Comment by petecooper** [comment]
Fixed in 
https://github.com/textpattern/textpacks/commit/e7fb8511e9c7d9000fbe00f3db4395973a81de52
